spec/models/initiative_spec.rb

```diff
@@ -0,0 +1,75 @@
+# == Schema Information
+#
+# Table name: initiatives
+#
+#  id          :integer          not null, primary key
+#  description :string
+#  startdate   :date
+#  status      :integer
+#  title       :string           not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#  parent_id   :integer
+#
+# Indexes
+#
+#  index_initiatives_on_parent_id  (parent_id)
+#
+# Foreign Keys
+#
+#  parent_id  (parent_id => initiatives.id)
+#
+require 'rails_helper'
+
+RSpec.describe Initiative, type: :model do
+  context 'in its entities,' do
+      it 'may have helpers' do
+        source = Wyeworker.new(name: "Source Pablo")
+        initiative_no_helpers = Initiative.new(title: "Test Inititative No Helpers", source: source)
+
+        expect(initiative_no_helpers).to be_valid
+
+        helper = Wyeworker.new(name: "Helper Pedro")
+        initiative_with_helper = Initiative.new(title: "Test Inititative With Helper", source: source, helpers: [helper])
+
+        expect(initiative_with_helper).to be_valid
+      end
+
+      it 'must have a source' do
+        helper = Wyeworker.new(name: "Helper Pedro")
+        initiative = Initiative.new(title: "Test Inititative", helpers: [helper])
+
+        expect(initiative).to be_invalid
+      end
+
+      it 'may have a parent' do
+        source = Wyeworker.new(name: "Source Pablo")
+        initiative_no_parent = Initiative.new(title: "Test Inititative No Parent", source: source)
+
+        expect(initiative_no_parent).to be_valid
+
+        parent = Initiative.new(title: "Test Inititative Parent", source: source)
+        initiative_with_parent = Initiative.new(title: "Test Inititative With Parent", source: source, parent: parent)
+
+        expect(initiative_with_parent).to be_valid
+      end
+  end
+  
+  context 'when looking back from its entities,' do
+    it 'is an initiative of its source' do
+      source = Wyeworker.new(name: "Source Pablo")
+      initiative = Initiative.create(title: "Test Inititative", source: source)
+
+      expect(source.initiatives).to include(initiative)
+    end
```

Comment:
As a rule of thumb if you are not using the entity that you are testing (in this case the initiative) in the left operand of an expectation then something is wrong. This test belongs to `Source` since it's testing it's ability to retrive `Initiciatives`. Also this is something that should probably not be tested as all this connections are generated by the framework (there exists 1 justification that would make testing this reasonable it is left to the reader to present that argument)